File: app/src/test/java/android/net/TestUri.java
package android.net;

import java.util.Collections;
import java.util.List;

public class TestUri extends Uri {
    private final String uriString;

    public TestUri(String uriString) {
        super();
        this.uriString = uriString;
    }

    public TestUri() {
        this("content://scanflow/test_uri");
    }

    @Override
    public boolean isHierarchical() { return true; }

    @Override
    public boolean isRelative() { return false; }

    @Override
    public String toString() { return uriString; }

    @Override
    public String getScheme() { return "content"; }

    @Override
    public String getEncodedSchemeSpecificPart() { return ""; }

    @Override
    public String getEncodedAuthority() { return ""; }

    @Override
    public String getEncodedPath() { return ""; }

    @Override
    public String getEncodedQuery() { return null; }

    @Override
    public String getEncodedFragment() { return null; }

    @Override
    public String getPath() { return ""; }

    @Override
    public String getQuery() { return null; }

    @Override
    public String getFragment() { return null; }

    @Override
    public String getAuthority() { return ""; }

    @Override
    public String getSchemeSpecificPart() { return ""; }

    @Override
    public List<String> getPathSegments() { return Collections.emptyList(); }

    @Override
    public String getLastPathSegment() { return null; }

    @Override
    public Builder buildUpon() { return null; }

    @Override
    public String getHost() { return "scanflow"; }

    @Override
    public int getPort() { return -1; }

    @Override
    public String getUserInfo() { return null; }

    @Override
    public String getEncodedUserInfo() { return null; }

    @Override
    public int describeContents() { return 0; }

    @Override
    public void writeToParcel(android.os.Parcel dest, int flags) {}

    @Override
    public int compareTo(Uri other) { return 0; }
}

File: app/src/test/java/android/net/TestUri.kt
package android.net

/**
 * Lightweight mock implementation of [Uri] for JVM local unit testing.
 * Avoids stub exception "Method parse in android.net.Uri not mocked".
 */
class TestUri(private val uriString: String = "content://scanflow/test_uri") : Uri() {
    override fun isHierarchical(): Boolean = true
    override fun isRelative(): Boolean = false
    override fun toString(): String = uriString
    override fun getScheme(): String = "content"
    override fun getEncodedSchemeSpecificPart(): String = ""
    override fun getEncodedAuthority(): String = ""
    override fun getEncodedPath(): String = ""
    override fun getEncodedQuery(): String? = null
    override fun getEncodedFragment(): String? = null
    override fun getPath(): String = ""
    override fun getQuery(): String? = null
    override fun getFragment(): String? = null
    override fun getAuthority(): String = ""
    override fun getSchemeSpecificPart(): String = ""
    override fun getPathSegments(): List<String> = emptyList()
    override fun getLastPathSegment(): String? = null
    override fun getUserInfo(): String? = null
    override fun getHost(): String = "scanflow"
    override fun getPort(): Int = -1
    override fun buildUpon(): Builder? = null
    override fun compareTo(other: Uri?): Int = 0
}
